src/model.cpp

```diff
@@ -221,6 +222,78 @@ size_t Model::unitsCount() const
     return mPimpl->mUnits.size();
 }
 
+void linkComponentVariableUnits(const ComponentPtr &component)
+{
+    for (size_t index = 0; index < component->variableCount(); ++index) {
+        auto v = component->variable(index);
+        auto u = v->units();
+        if (u != nullptr) {
+            auto model = owningModel(u);
+            if (model == nullptr && !isStandardUnitName(u->name())) {
+                model = owningModel(component);
+                auto modelUnits = model->units(u->name());
+                if (modelUnits == nullptr) {
+                    model->addUnits(u);
+                    modelUnits = u;
+                }
+                v->setUnits(modelUnits);
+            }
+        }
+    }
+}
+
+void traversComponentTreeLinkingUnits(const ComponentPtr &component)
```

Comment:
I want to add an 'e' to the `travers`...

Reply:
Done.